source/common/stats/stat_merger.cc

```diff
@@ -63,15 +60,35 @@ void StatMerger::mergeCounters(const Protobuf::Map<std::string, uint64_t>& count
 
 void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges) {
   for (const auto& gauge : gauges) {
-    auto& gauge_ref = temp_scope_->gauge(gauge.first);
+    StatNameManagedStorage storage(gauge.first, temp_scope_->symbolTable());
+    StatName stat_name = storage.statName();
+    absl::optional<std::reference_wrapper<const Gauge>> gauge_opt =
+        temp_scope_->findGauge(stat_name);
+
+    // We the stat named in the protobuf map is already initialized, and has a mode
```

Comment:
nit: grammar

Reply:
thanks; fixed;. I think the "we" should just be an "if".